### ISSUE TITLE
the mp4 don't play perfectly if turned of 90 degree 1080x1920

### ISSUE THREAD

**Issue by matteoraggi** [ogra1/omxplayer-snap#4]
It was not easy to launch the mp4 like with raspbian, but at the end I was able to launch omxplayer-pi on Ubuntu Core 18.04 on Raspberry PI 3 model B.
And it works perfectly with mp4 1920x1080!
But then turning a mp4 of aspect ratio 1080x1920 of 90 degree, then it looks that the fps is lower of 30fps (even if trying to force it with the --fps parameter) and that there are some white pixel disturb on a frame... Do you think to can patch it in a short time? Omxplayer on raspbian don't have this issue, I tested it many times..

**Comment by ogra1** [comment]
have you tried rotating the display from the Pi's config.txt file instead (and let omxplayer just play without rotation option) ?

**Comment by matteoraggi** [comment]
Originally I was rotating all the pi's operating system, but then it was giving a visualization issue:
https://github.com/popcornmix/omxplayer/issues/553
That I skipped exactly turning omxplayer of 90 degree instead to to turn the operating systme of 90 degree.

**Comment by ogra1** [comment]
I just tested rotated playback on an Ubuntu Core18 system using the omxplayer-pi snap from the beta and edge channels just using a livestream:

omxplayer-pi --orientation 90 https://rbmn-live.akamaized.net/hls/live/2002830/geoSTVDEweb/master_6692.m3u8

It could be related to the fact that Core images are defaulting to dtoverlay=vc4-fkms-v3d (or -kms-v3d, I don't remember from the top of my head) which is needed for mir-kiosk to offer wayland support. 

When commenting the dtoverlay line in config.txt I get proper playback with no stuttering or visual issues ... watching the system via htop from an additional terminal shows no significant cpu use at all.

**Comment by matteoraggi** [comment]
Thanks, I already removed the driver that you suggest to remove 1-2 days ago, and now doing it:

> omxplayer-pi --orientation 90 https://rbmn-live.akamaized.net/hls/live/2002830/geoSTVDEweb/master_6692.m3u8

it says 

> have a nice day

on Raspbian when I try to wget it I get 403 Forbidden and from chrome windows 10 it answer: 

> access denied

the omxplayer on Raspbian also says have a nice day, if launching it:

> omxplayer --orientation 90 https://rbmn-live.akamaized.net/hls/live/2002830/geoSTVDEweb/master_6692.m3u8

I have the issue playing simple mp4 like this one demo that I uploaded for you: http://www.copteros.com/portait1080x1920demo.mp4

**Comment by ogra1** [comment]
i just noticed that upstream had some new commits, when doing a local build with the latest code and playing it like:
```
ogra@localhost:~$ omxplayer-pi --orientation 90 http://www.copteros.com/portait1080x1920demo.mp4
Video codec omx-h264 width 1080 height 1920 profile 77 fps 24.000000
Subtitle count: 0, state: off, index: 1, delay: 0
Error: Unable to open font
V:PortSettingsChanged: 1080x1920@24.00 interlace:0 deinterlace:0 anaglyph:0 par:1.00 display:0 layer:0 alpha:255 aspectMode:0
have a nice day ;)
ogra@localhost:~$
```
properly shows an "alice in wonderland" scene in portrait, seemingly with no degraded fps and with no excessive cpu usage.

a build with latest upstream code is currently running on build.snapcraft.io, once this is built you should be able to "snap refresh omxplayer-pi --edge" to get it and to try if this helps with your issue.

**Comment by matteoraggi** [comment]
You are right, I gave you the wrong file, this one plays perfect to me too:
http://www.copteros.com/portait1080x1920demo.mp4
I tryed on ubuntu server 18.04.02 for raspberry 
http://cdimage.ubuntu.com/ubuntu/releases/bionic/release/ubuntu-18.04.2-preinstalled-server-armhf+raspi3.img.xz
and I still get the same issue, but **please give one more test with this video in loop**, this one is givign the issue, probably is the fading giving the problem with 256mb of gpu:
http://185.24.234.250//single_slide_videos//df3a1d7d08de92fb7a82ab5283624c71.mp4

**Comment by ogra1** [comment]
> I tryed on ubuntu server 18.04.02 for raspberry

Ah, you said Ubuntu Core 18 above, Ubuntu Server could be on a different firmware level ...

**Comment by matteoraggi** [comment]
yes, I tryed on both the ubuntu, core and server and the final result is the same, just alice is playing well instead the other video with fading is missing many frames..
